Dfe.Data.SearchPrototype/Web/Tests/Integration/PageObjectModel/PageComponents/PageHeader.cs

```diff
@@ -0,0 +1,43 @@
+﻿using AngleSharp.Dom;
+using AngleSharp.Html.Dom;
+using DfE.Data.SearchPrototype.Web.Tests.Integration.PageObjectModel.Setup;
+using Microsoft.AspNetCore.Mvc.Testing;
+
+namespace DfE.Data.SearchPrototype.Web.Tests.Integration.PageObjectModel.PageComponents
+{
+    public class PageHeader : PageObjectModelExtractor
+    {
+        public PageHeader(WebApplicationFactory<Program> webApplicationFactory) :
+            base(webApplicationFactory){
+        }
+
+        public async Task<string> GetHeading()
```

Comment:
Just wondering if we need a function for each element on a page as we're doing here?
In our page integration tests we utilise Selenium to do the web element capturing and existence checking for us.
An example of a page object from the SD suite: [SearchPage.cs](https://agilefactory.visualstudio.com/SPT/_git/DfE.Data.PerformanceTables.Web?path=/tests/DfE.Data.PerformanceTables.Web.UI.SharedTests/Pages/Search/SearchPage.cs&version=GBmain&line=10&lineEnd=11&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents)
And the respective tests (using Fluent Assertions for the checks): [SearchTests.cs](https://agilefactory.visualstudio.com/SPT/_git/DfE.Data.PerformanceTables.Web?path=/tests/DfE.Data.PerformanceTables.Web.UI.PageIntegrationTests/Tests/Pages/Search/SearchSchoolCollegeTests.cs&version=GBmain&line=24&lineEnd=25&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents)

Reply:
We do, they're encapsulated in the page component(s) which the given page will be composed of. We have page header as an example, which is part of the home page composition, once we build out each page will be enriched with more page components which will encapsulate the underlying behaviour (functions).  As such, we can compose our tests to use HomePage.PageHeader.GetPageHeading() for example.